src/components/Box.js

```diff
@@ -91,12 +128,7 @@ const Box = forwardRef(
           extend
         )}>
         {space
-          ? Children.toArray(children).map((child, index, arr) => (
-              <React.Fragment key={index}>
-                {child}
-                {index === arr.length - 1 ? null : <Spacer size={space} />}
-              </React.Fragment>
-            ))
+          ? renderChildren(children, spaceType, space, direction)
```

Comment:
Not that it matters too much, but you could have a component to do that instead of a function.

```js
<SpaceOut type={spaceType} space={space} direction={direction}>{children}</SpaceOut>